packages/embedded-wallet/src/json-rpc-router/JsonRpcRoute.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createFactory } from "react";
+import { JsonRPCRequest } from "web3/providers";
+import { JsonRpcRouterConsumer } from "./JsonRpcRouter";
+
+export type JsonRpcComponentProps = { request: JsonRPCRequest };
+export type JsonRpcRouteProps = {
+  method: string;
+  component: React.ComponentClass<JsonRpcComponentProps> | React.FunctionComponent<JsonRpcComponentProps>;
+};
+
+/**
+ * Declares a listener component for a given JSON-RPC method.
+ *
+ * @example
+ * ```
+ * <JsonRpcRoute method="jsonrpc_doSomething" component={DoSomethingUI} />
+ * ```
+ *
+ * @property method - The name of the JSON-RPC method.
+ * @property component - A React component that will be rendered as a response to the message.
+ */
+const JsonRpcRoute: React.FC<JsonRpcRouteProps> = ({ method, component }: JsonRpcRouteProps) => {
+  return (
+    <JsonRpcRouterConsumer>
```

Comment:
should the responsibility of defining which component to render fall back to the routes, or to the router?.
Correct me if I'm wrong, but this logic check shouldn't be on the router instead of here?

Reply:
In the declaration, the Route defines which method goes with each component. The router would have to be inspecting each child element instead of delegating and letting them resolve whether to render or not on their own.

Take a look at [what `react-router` does](https://github.com/ReactTraining/react-router/blob/master/packages/react-router/modules/Route.js#L38-L42). Granted, they have a `Switch` component that does the logic check, but they _also_ check for the match locally if it hasn't been computed somewhere else.